### PR TITLE
Ability to copy character codepoints

### LINKED ISSUE #48: [Feature request] Option to show Unicode character codepoint(s)
Often times I'll need to look up the specific codepoint of an emoji or other character online and it would be much easier to just have it in the menu. I'm thinking it could be dynamically inserted into the entry text, like so:

- 😀 `<small>1F600</small> grinning face <small>(face, grin, grinning face)</small>`
- 🧑‍💻 `<small>1F9D1+200D+1F4BB</small> technologist <small>(coder, developer, inventor, software, technologist)</small>`

There are some considerations for implementing this:
- Codepoints are not stored in the character files so they would need to be generated on-the-fly.
- Should it omit invisible joiners? They're important to know sometimes.
- Can it accurately insert the text in the right spot or should it be attached to an end?

### PR DESCRIPTION
**Pull request by katacarbix**
Closes #48
I understand you're doing a bit of refactoring in another branch. Take your time with reviewing this! 😁

### CODE REVIEW

**Comment by fdw** [comment]
Would it be possible (and sensible for you) to not show the codepoint, but only calculate it if the user presses `Alt+u`? This would be simpler than the current way with an additional cli parameter.

**Comment by katacarbix** [comment]
Sure, that sounds reasonable! I think there should still be a shortcut to copy it to the clipboard; what would make the most sense to you?

**Comment by fdw** [comment]
If we can find a good key for it, why not? I'm thinking of `i`, as it is right next to `u` (on most keyboards, at least). Do you have a better idea?

**Comment by katacarbix** [comment]
Works for me! I'll try to finish it up over the weekend.

**Comment by katacarbix** [comment]
Done. Updated the readme as well.

**Comment by fdw** [comment]
Sorry, I'm confused now: I thought `Alt+i` should copy the codepoint to the clipboard (probably using `copy_to_clipboard`), but in your current version it's used to show the codepoints in the main screen.

**Comment by katacarbix** [comment]
Oh you're right, my bad!!

**Comment by fdw** [comment]
Sorry, let me rephrase: I'm not too convinced on a shortcut that shows the codepoint. One to directly enter it (`Alt+u`), and one to copy (`Alt+i`) make sense to me, but - going from your usecase - I didn't think that showing all of them is necessary.

**Comment by katacarbix** [comment]
I see what you mean, sorry for the confusion. I think that would be the safer route.

For some reason I cannot get `type_characters` to work (it causes my WM to behave erratically) but `copy_paste_characters` does. Obviously this is not ideal, but I'll push what I have and maybe someone else can figure out what's going on.

**Comment by fdw** [comment]
Thaks, that looks very clean now 🙂

What were your problems with `type_characters`? Into which program did you try to type?

**Comment by katacarbix** [comment]
Oh, I think I figured out what's happening. `xdotool` is typing the characters in the split second after I press the shortcut key and before I release `alt`. `alt` is my i3 mod key so it interprets those keystrokes as commands to fullscreen/switch workspaces/etc.

Maybe we could add a short delay? It probably wouldn't work if I kept holding `alt`, though. The solution is probably just to set `insert-with-clipboard` to true on your own machine and make a note somewhere in the documentation about it.

**Comment by fdw** [comment]
Hey, thanks for finding that out!

I'm not sure how to best do it, but I think adding a delay would be a good idea. `rofimoji` should already use `xdotool` with `--clearmodifiers`, but I have no idea what happens if you really release `alt` while your system thinks it has been released previously.

Maybe we can add some small delay and test if it is noticable?

**Comment by katacarbix** [comment]
It looks like `xdotool type` isn't being passed the `--clearmodifiers` flag. With that added it works as it should!

https://github.com/fdw/rofimoji/blob/a69104b938b5576691c5022c0012b41bb073062f/picker/rofimoji.py#L363-L370

File: picker/rofimoji.py [hunk 1]
```diff
@@ -153,11 +157,11 @@ def read_character_files(file_names: List[str]) -> str:
 
     for file_name in file_names:
         entries = entries + load_from_file(file_name)
-
+    
     return entries
 
 
-def resolve_all_files(file_names):
+def resolve_all_files(file_names: List[str]) -> List[str]:
```
Comment (fdw) [inline picker/rofimoji.py:164]: Thanks!

**Review by fdw — APPROVED** [review]
